### ISSUE TITLE
Miki Error Handling

### ISSUE THREAD

**Issue by LovingMelody** [veldlabs/miki-bot#151]
Error Handling:
- [ ]  Permission Errors
- [ ] ???

Permission Errors:
Currently, Miki will throw an error on the back-end if Miki receives an error from lack of permissions, this isn't appropriate. This should be sent to the server and suppressed on the back-end, if an error occurs from lack no permission to send message, silently ignore. It would probably be best to add a check to commands to verify the permissions are met before executing the command.

**Comment by velddev** [comment]
🤔 Yeah. That's the idea, this new system is used for uncaught errors however.

**Comment by LovingMelody** [comment]
There is really no reason why a lot of these are uncaught.

**Comment by velddev** [comment]
They are uncaught because they used to be set to the standard error message

But this made it really hard for me to track bugs

**Comment by LovingMelody** [comment]
@velddev 
![image](https://cloud.githubusercontent.com/assets/17859309/26511666/0c93dc54-4231-11e7-8ffc-3fb61de81c97.png)
seems to work great

**Comment by velddev** [comment]
Marriage should work correctly now. Error handling is more smooth now with the missing checks being filled in

**Comment by LovingMelody** [comment]
This is not yet solved

**Comment by velddev** [comment]
Here is my idea for this issue, internally you set up responses to exceptions, and whenever someone gets into a bad thing e.g. not having an account you throw "UserNullException" e.g.

```
public async Task SomeCommandAsync(EventContext e) {
   User user = await User.GetAsync(e.Author);
   if (user == null) 
      throw new UserNullException();
   ...
}

**Comment by velddev** [comment]
Update on this: The update on this is planned to be implemented in the Miki.Framework version shipped with `0.6.1`, but the implementation will start being enforced at `0.7`

**Comment by velddev** [comment]
This has been added to Miki, not all routes have been done yet. but for now I will close this issue. read more about this in https://github.com/Mikibot/Miki.Framework/blob/master/Miki.Framework/Events/EventSystem.cs#L67
